src/cmd/singularity/cli/build.go

```diff
@@ -102,6 +102,8 @@ var BuildCmd = &cobra.Command{
 					authWarning != auth.WarningTokenTooShort {
```

Comment:
This should probably be refactored? Don't think you need to test the value of authWarning against all of these? Plus  the nested if / else are unnecessarily complex.